barretenberg/cpp/src/barretenberg/stdlib/transcript/transcript.hpp

```diff
@@ -19,7 +19,23 @@ template <typename Builder> struct StdlibTranscriptParams {
         Builder* builder = data[0].get_context();
         return stdlib::poseidon2<Builder>::hash(*builder, data);
     }
-
+    /**
+     * @brief Split a challenge field element into two half-width challenges
+     * @details `lo` is 128 bits and `hi` is 126 bits.
+     * This should provide significantly more than our security parameter bound: 100 bits
+     *
+     * @param challenge
+     * @return std::array<Fr, 2>
+     */
+    static inline std::array<Fr, 2> split_challenge(const Fr& challenge)
```

Comment:
not too familiar with cycle_scalar... why are we using it instead of normal field_ts?

Reply:
cycle_scalar has an existing method to split a stdlib::field_t element into two half-width scalars. It's nontrivial to do efficiently so I wanted to reuse existing code

Reply:
it's a little unfortunate we have to bring in cycle group (which shouldn't have anything to do with this) but alright. I guess I don't have a better solution beyond copy paste... or some bigger refactoring